src/linter.cc

```diff
@@ -57,140 +59,162 @@ absl::Status PrintASTTree(absl::string_view sql) {
-          absl::StrCat("Lines should be <= ", std::to_string(line_limit),
-                       " characters long [", std::to_string(line_number),
-                       ",1]"));
+
+    if (lineSize > line_limit && line_number > last_added) {
+      last_added = line_number;
+      result.Add(ErrorCode::kLineLimit, sql, i,
+                 absl::StrCat("Lines should be <= ", std::to_string(line_limit),
+                              " characters long"));
     }
   }
-  return absl::OkStatus();
+  return result;
 }
 
-absl::Status CheckParserSucceeds(absl::string_view sql) {
+LinterResult CheckParserSucceeds(absl::string_view sql) {
   std::unique_ptr<ParserOutput> output;
 
   ParseResumeLocation location = ParseResumeLocation::FromStringView(sql);
 
   bool is_the_end = false;
+  int byte_position = 1;
+  LinterResult result;
   while (!is_the_end) {
-    ZETASQL_RETURN_IF_ERROR(ParseNextScriptStatement(&location, ParserOptions(),
-                                                     &output, &is_the_end));
+    byte_position = location.byte_position();
+    absl::Status status = ParseNextScriptStatement(&location, ParserOptions(),
+                                                   &output, &is_the_end);
+    if (!status.ok()) {
+      // TODO(orhanuysal): Implement a token parser to seperate statements
+      // currently, when parser fails, it is unable to determine
+      // the end of the statement.
+      result.Add(ErrorCode::kParseFailed, sql, byte_position, "Parser Failed");
+      break;
+    }
   }
-  return absl::OkStatus();
+  return result;
 }
 
-absl::Status CheckSemicolon(absl::string_view sql) {
+LinterResult CheckSemicolon(absl::string_view sql) {
+  LinterResult result;
   std::unique_ptr<ParserOutput> output;
 
   ParseResumeLocation location = ParseResumeLocation::FromStringView(sql);
   bool is_the_end = false;
 
   while (!is_the_end) {
-    ZETASQL_RETURN_IF_ERROR(ParseNextScriptStatement(&location, ParserOptions(),
-                                                     &output, &is_the_end));
-
+    absl::Status status = ParseNextScriptStatement(&location, ParserOptions(),
+                                                   &output, &is_the_end);
+    if (!status.ok()) return LinterResult(status);
     int location =
         output->statement()->GetParseLocationRange().end().GetByteOffset();
 
     if (location >= sql.size() || sql[location] != ';') {
-      return absl::Status(absl::StatusCode::kFailedPrecondition,
-                          "Each statemnt should end with a consequtive"
-                          "semicolon ';'");
+      result.Add(ErrorCode::kSemicolon, sql, location,
+                 "Each statemnt should end with a consequtive"
+                 "semicolon ';'");
     }
   }
-  return absl::OkStatus();
+  return result;
 }
 
-bool AllUpperCase(const absl::string_view &sql,
-                  const ParseLocationRange &range) {
+bool ConsistentUppercaseLowercase(const absl::string_view &sql,
```

Comment:
Similar to comments check, I think we should a config that says what exactly is prohibited. So that a project that uses ALL CAPS will not accept a file that uniformly uses only lower case. Please, open an issue to fix it later.

Reply:
Sure. Then I assume there will be users who set the config as all lowercase and their lint will not allow keywords that uniformly consists of uppercase letters.

Reply:
Ok!